triton-cli/src/main.rs

```diff
@@ -0,0 +1,210 @@
+    public_outputs: Vec<u64>,
+    proof: Vec<u64>,
+}
+
+fn read_proof(proof_path: &str) -> Result<(Stark, Claim, Proof)> {
+    let mut file = fs::File::open(proof_path)?;
+    let mut proof_bytes: Vec<u8> = Vec::new();
+    file.read_to_end(&mut proof_bytes)?;
+    let serialized_proof: SerializedProof = bincode::deserialize(&proof_bytes)?;
+    let stark = Stark {
+        security_level: usize::try_from(serialized_proof.security_level)?,
+        fri_expansion_factor: usize::try_from(serialized_proof.fri_expansion_factor)?,
+        num_trace_randomizers: usize::try_from(serialized_proof.num_trace_randomizers)?,
+        num_collinearity_checks: usize::try_from(serialized_proof.num_colinearity_checks)?,
+        num_combination_codeword_checks: usize::try_from(
+            serialized_proof.num_combination_codeword_checks,
+        )?,
+    };
+    let mut digest = Digest::default();
+    assert_eq!(
+        digest.0.len(),
+        serialized_proof.program_digest.len(),
+        "digest length mismatch!"
+    );
+    serialized_proof
+        .program_digest
+        .iter()
+        .enumerate()
+        .for_each(|(i, x)| digest.0[i] = BFieldElement::new(*x));
+    let claim = Claim {
+        program_digest: digest,
+        input: serialized_proof
+            .public_inputs
+            .iter()
+            .map(|v| BFieldElement::new(*v))
+            .collect(),
+        output: serialized_proof
+            .public_outputs
+            .iter()
+            .map(|v| BFieldElement::new(*v))
+            .collect(),
+    };
+    let proof_vec = serialized_proof
+        .proof
+        .iter()
+        .map(|v| BFieldElement::new(*v))
+        .collect();
+    Ok((stark, claim, Proof(proof_vec)))
+}
+
+fn write_proof(data: (Stark, Claim, Proof), out: &str) -> Result<()> {
+    let (stark, claim, proof) = data;
+    let serialized = SerializedProof {
+        version: 1,
+        security_level: u64::try_from(stark.security_level)?,
+        fri_expansion_factor: u64::try_from(stark.fri_expansion_factor)?,
+        num_trace_randomizers: u64::try_from(stark.num_trace_randomizers)?,
+        num_colinearity_checks: u64::try_from(stark.num_collinearity_checks)?,
+        num_combination_codeword_checks: u64::try_from(stark.num_combination_codeword_checks)?,
+        program_digest: claim.program_digest.0.iter().map(u64::from).collect(),
+        public_inputs: claim.input.iter().map(u64::from).collect(),
+        public_outputs: claim.output.iter().map(u64::from).collect(),
+        proof: proof.0.iter().map(u64::from).collect(),
+    };
+    let mut file = fs::File::create_new(out)?;
+    let proof_bytes = bincode::serialize(&serialized)?;
+    file.write_all(&proof_bytes)?;
+    Ok(())
+}
+
+#[test]
+fn test_serialization() -> Result<()> {
+    let asm = "./test-vectors/simple.tasm".to_string();
+    let proof = "./test-vectors/simple.proof".to_string();
+    prove(&asm, &proof, None, None)?;
+    verify(&proof);
```

Comment:
Nit: you can leave the strings on the stack.

```suggestion
    let asm = "./test-vectors/simple.tasm";
    let proof = "./test-vectors/simple.proof";
    prove(asm, proof, None, None)?;
    verify(proof);
```